### PR TITLE
Get brave-ui directly from npm via @brave/brave-ui and don't compile typescript in node_modules

### PR DESCRIPTION
**Pull request by petemill**
Resolves https://github.com/brave/brave-browser/issues/32092

There should be no observable change in the browser or build.

Published version of `@brave/brave-ui` includes https://github.com/brave/brave-ui/pull/642 with JS output

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by josheleonard — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -277,6 +277,7 @@
   "devDependencies": {
     "@babel/core": "7.16.0",
     "@babel/plugin-proposal-export-namespace-from": "7.16.0",
+    "@brave/brave-ui": "0.40.1",
```
Comment (diracdeltas) [inline package.json:280]: should this actually be in deps instead of devdeps?

**Review by fallaciousreasoning — APPROVED** [review]
LGTM!

File: package.json (hunk 1 above)
Comment (fallaciousreasoning) [reply package.json:280]: I wonder if the old one was a devDependency because we were compiling it? Not sure though

Comment (petemill) [reply package.json]: I suppose the only(?) difference between devDependencies and dependencies for brave-core is whether we check for a LICENSE file during build. So in that sense brave-ui should be in dependencies.  Changed

File: components/brave_wallet_ui/components/shared/locale-restricted/locale-restricted.test.tsx [hunk 2]
```diff
@@ -7,11 +7,15 @@ import { shallow } from 'enzyme'
...
+const currentLocale = Intl.NumberFormat().resolvedOptions().locale
+
+
 describe('<LocaleRestricted />', () => {
   it('renders children when user\'s locale is allowed', () => {
+    expect(currentLocale.length).toBeGreaterThanOrEqual(1)
     const wrapper = shallow((
       <LocaleRestricted
-        allowedLocales={['en-US']}
+        allowedLocales={[currentLocale]}
```
Comment (petemill) [inline components/brave_wallet_ui/components/shared/locale-restricted/locale-restricted.test.tsx:18]: I needed to change this as it seems this test doesn't pass if the local machine is not en_US locale. I guess we're just getting lucky @josheleonard?

**Comment by petemill** [comment]
Updated @brave/brave-ui to fix issue in jest which needed commonJS format and not esmodule for JS files

File: components/common/typescript.gni [hunk 3]
```diff
@@ -45,7 +45,8 @@ brave_common_web_compile_inputs = [
 
   # brave-ui changes (perhaps a manual local link which wouldn't affect
   # //brave/package.json)
-  "//brave/node_modules/brave-ui/package.json",
+  "//brave/node_modules/@brave/brave-ui/package.json",
+  "//brave/node_modules/@brave/leo/package.json",
```
Comment (fallaciousreasoning) [inline components/common/typescript.gni:49]: Do we need to add Leo here? It seems kind of unrelated to the other changes? I also don't think we need to compile any Leo stuff with Typescript (but I could be wrong :laughing:)

File: jest.config.js [hunk 4]
```diff
@@ -74,8 +74,8 @@ module.exports = {
     "\\.(css|less|scss)$": "identity-obj-proxy",
     "^\\$web-common\\/(.*)": "<rootDir>/components/common/$1",
     "^\\$web-components\\/(.*)": "<rootDir>/components/web-components/$1",
-    "^brave-ui$": "<rootDir>/node_modules/brave-ui/src",
-    "^brave-ui\\/(.*)": "<rootDir>/node_modules/brave-ui/src/$1",
+    "^brave-ui$": "<rootDir>/node_modules/@brave/brave-ui",
+    "^brave-ui\\/(.*)": "<rootDir>/node_modules/@brave/brave-ui/$1",
```
Comment (fallaciousreasoning) [inline jest.config.js:78]: Don't need to do it here, but I wonder if we should get rid of these import maps. I don't really think they add anything :laughing:

**Review by fallaciousreasoning — APPROVED** [review]
Still LGTM

File: components/common/typescript.gni (hunk 3 above)
Comment (petemill) [reply components/common/typescript.gni:49]: It's more to make sure that WebUIs get re-compiled with webpack when Leo changes are made. It's not typescript related. Granted, those changes would have to be made without changing the brave/package.json file. It would have to be an npm install without a save. But if we're doing that strategy for brave-ui then we should do it for leo IMO.

Comment (petemill) [reply components/common/typescript.gni:49]: Actually, I keep forgetting that _all_ dependency modules will be included in the generated *.d "dep" files, including files in node_modules, so if any of those change, and they're used in a target, then that target will be recompiled.

File: jest.config.js (hunk 4 above)
Comment (petemill) [reply jest.config.js:78]: This one is needed to map brave-ui to @brave/brave-ui, or do you mean something else?

**Comment by petemill** [comment]
One build is failing due to unrelated cargo dependency

```
[2023-08-14T19:57:32.495Z] Auditing (cargo) third_party/rust

[2023-08-14T19:57:32.495Z]     Fetching advisory database from `[https://github.com/RustSec/advisory-db.git`](https://github.com/RustSec/advisory-db.git%60)

[2023-08-14T19:57:32.495Z]       Loaded 557 security advisories (from /home/ubuntu/.cargo/advisory-db)

[2023-08-14T19:57:32.495Z]     Updating crates.io index

[2023-08-14T19:57:32.495Z]     Scanning /home/ubuntu/workspace/pr-brave-browser-brave-ui-npm-noplatform/src/brave/third_party/rust/Cargo.lock for vulnerabilities (313 crate dependencies)

[2023-08-14T19:57:32.495Z] Crate:     ed25519-dalek

[2023-08-14T19:57:32.495Z] Version:   1.0.1

[2023-08-14T19:57:32.495Z] Title:     Double Public Key Signing Function Oracle Attack on `ed25519-dalek`

[2023-08-14T19:57:32.495Z] Date:      2022-06-11

[2023-08-14T19:57:32.495Z] ID:        RUSTSEC-2022-0093

[2023-08-14T19:57:32.495Z] URL:       https://rustsec.org/advisories/RUSTSEC-2022-0093

[2023-08-14T19:57:32.495Z] Solution:  Upgrade to >=2

[2023-08-14T19:57:32.495Z] Dependency tree:

[2023-08-14T19:57:32.495Z] ed25519-dalek 1.0.1

[2023-08-14T19:57:32.495Z] └── ed25519-dalek-bip32 0.2.0

[2023-08-14T19:57:32.495Z]     └── chromium 0.1.0

[2023-08-14T19:57:32.495Z]
```

File: jest.config.js (hunk 4 above)
Comment (fallaciousreasoning) [reply jest.config.js:78]: Yeah, exactly - it just seems like an unnecessary bit of indirection. But not that important, and I don't think we need to do anything here.

Comment (petemill) [reply jest.config.js:78]: Without it, we'd need to find/replace all uses of 'brave-ui/xyz' with '@brave/brave-ui/xyz' - it's similar to the ones we have for webpack and typescript.

File: components/common/typescript.gni (hunk 3 above)
Comment (petemill) [reply components/common/typescript.gni:49]: removed them

**Review by diracdeltas — APPROVED** [review]
